### PR TITLE
Adding mpi tests to all unit tests

### PR DESCRIPTION
**Pull request by timryanb**
Fixed broken parallel tests

### CODE REVIEW

File: src/constitutive/TACSIsoShellConstitutive.cpp [hunk 1]
```diff
@@ -321,40 +329,49 @@ TacsScalar TACSIsoShellConstitutive::evalFailureStrainSens( int elemIndex,
...
+    ksFactor = exp(ksWeight*(bottom - ksMax)) / ksSum;
 
-      sens[6] = sens[7] = sens[8] = 0.0;
+    sens[0] += ksFactor*phi[0];
+    sens[1] += ksFactor*phi[1];
+    sens[2] += ksFactor*phi[2];
 
-      return bottom;
-    }
+    sens[3] += ksFactor*-ht*phi[0];
```
Comment (gjkennedy) [inline src/constitutive/TACSIsoShellConstitutive.cpp:368]: the *-ht* is a bit confusing here.. can we just make it
sens[3] -= ksFactor*ht*psi[0];

File: src/elements/TACSElement.cpp [hunk 2]
```diff
@@ -233,9 +245,23 @@ void TACSElement::addAdjResProduct( int elemIndex, double time,
       product += psi[i]*TacsImagPart(tmp[i])/dh;
     }
 #else
-    for ( int i = 0; i < nvars; i++ ){
-      product += psi[i]*(tmp[i] - res[i])/dh;
-    }
+      if (fdOrder < 2){
```
Comment (gjkennedy) [inline src/elements/TACSElement.cpp:248]: Indentation looks off here? 2-spaces only?

File: src/elements/TACSElement.cpp [hunk 3]
```diff
@@ -20,6 +20,18 @@
...
 
+/*
+  Default finite difference order for real analysis
+*/
+int TACSElement::fdOrder = 2;
+
+/*
+  Allow users to set default finite difference order for real analysis
+*/
+void TACSElement::setFDOrder( int order ){
```
Comment (gjkennedy) [inline src/elements/TACSElement.cpp:31]: what about the full name setFiniteDifferenceOrder()?

File: src/elements/TACSElement.cpp (hunk 2 above)
Comment (timryanb) [reply src/elements/TACSElement.cpp]: Should be addressed in last commit

File: src/constitutive/TACSIsoShellConstitutive.cpp (hunk 1 above)
Comment (timryanb) [reply src/constitutive/TACSIsoShellConstitutive.cpp]: Should be addressed in last commit

**Review by gjkennedy — APPROVED** [review]

**Pull request merged**